app/controllers/organizations/adoptable_pets_controller.rb

```diff
@@ -55,6 +51,8 @@ def set_likes
 
     def set_species
       @species = params[:species]
+
+      redirect_back_or_to root_path if @species.nil?
```

Comment:
I prefer to keep the action method happy path only, so moving this here

Reply:
Yeah it took me a minute to get used to this approach but now I really like it. Keeping the action happy path only and handling edges in other methods. It's a way to keep controllers very clean. I would counter and say that it's really not two different responsibilities. The method is setting the species, and handling the event that it is nil accordingly.